ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/plugins/api/ApplicationPluginBuilder.kt

```diff
@@ -82,151 +50,175 @@ public abstract class PluginBuilder<PluginConfig : Any> internal constructor(
+     *
+     * @param block An action that needs to be executed when your server is sending a response to a client.
+     **/
+    public override fun onCallRespond(
+        block: suspend CallRespondContext<PluginConfig>.(call: ApplicationCall, body: Any) -> Unit
+    ) {
+        onDefaultPhase(
+            onResponseInterceptions,
+            ApplicationSendPipeline.Transform,
+            PHASE_ON_CALL_RESPOND,
+            ::CallRespondContext,
+            block
+        )
     }
 
     /**
      * Executes specific actions after all [targetPlugins] are executed.
      *
-     * @param targetPlugins Plugins that need to be executed before your current [PluginBuilder].
+     * @param targetPlugins Plugins that need to be executed before your current [ApplicationPluginBuilder].
      * @param build Defines the code of your plugin that needs to be executed after [targetPlugins].
      *
      * Note: you can define multiple actions inside a [build] callback for multiple stages of handling an HTTP call
      * (such as [onCall], [onCallRespond], and so on). These actions are executed right after all actions defined
      * by the given [plugin] are already executed in the same stage.
      **/
-    public fun afterPlugins(
+    public fun after(
         vararg targetPlugins: Plugin<*, *, PluginInstance>,
-        build: AfterPluginBuilder.() -> Unit
+        build: AfterPluginsBuilder<PluginConfig>.() -> Unit
     ) {
         pipelineHandlers.add { pipeline ->
-            AfterPluginBuilder(this, targetPlugins.map { pipeline.plugin(it).builder }).build()
+            AfterPluginsBuilder(this, targetPlugins.map { pipeline.plugin(it).builder }).build()
         }
     }
 
     /**
      * Executes specific actions before all [targetPlugins] are executed.
      *
-     * @param targetPlugins Plugins that need to be executed after your current [PluginBuilder].
+     * @param targetPlugins Plugins that need to be executed after your current [ApplicationPluginBuilder].
      * @param build Defines the code of your plugin that needs to be executed before [targetPlugins].
      *
      * Note: you can define multiple actions inside a [build] callback for multiple stages of handling an HTTP call
      * (such as [onCall], [onCallRespond], and so on) and each of these actions will be executed right before all actions defined
      * by the given [targetPlugins] were already executed in the same stage.
      **/
-    public fun beforePlugins(
+    public fun before(
         vararg targetPlugins: Plugin<*, *, PluginInstance>,
-        build: BeforePluginsBuilder.() -> Unit
+        build: BeforePluginsBuilder<PluginConfig>.() -> Unit
     ) {
         pipelineHandlers.add { pipeline ->
             BeforePluginsBuilder(this, targetPlugins.map { pipeline.plugin(it).builder }).build()
         }
     }
 
+    public fun <EventConfig> on(
+        event: Event<EventConfig>,
+        config: EventConfig
```

Comment:
Already discussed with you that I would much more prefer the following naming here:
```
public fun <Handler> on(
        event: Event<EventConfig>,
        handle: Handler       
}
```

Because `handler` to me is a function (something that can be executed at some point when event happens) while `config` is a data class containing some configuration data.
If you insist on the "config..." word maybe let's change to `configure` because it's a verb?